### PR TITLE
Board detect + assorted fixes

### PR DESCRIPTION
**Pull request by mbelluzzo**

### CODE REVIEW

File: src/lib/common/Kconfig [hunk 1]
```diff
@@ -8,7 +8,7 @@ config BOARD_NAME
 
             Usually the board name is detected using either
             SOL_BOARD_NAME environment variable or instructions from
-            'board_detect.json' if DETECT_BOARD_NAME is enabled. Both
+            '99_board_detect.json' if DETECT_BOARD_NAME is enabled. Both
```
Comment (barbieri) [inline src/lib/common/Kconfig]: sed too much? I don't think this is to say "99" in there

File: src/lib/common/sol-board-detect.c [hunk 2]
```diff
@@ -207,8 +207,31 @@ _json_open_doc(const char *path, struct sol_json_scanner *scanner)
...
+    int r;
+    char *file_path;
+
+    if (!ent) //Just skip, no need to trigger a warning.
+        return false;
+
+    if (ent->d_type != DT_REG && ent->d_type != DT_LNK) //It won't be recursive
+        return false;
+
+    if (!ent->d_name || !ent->d_name[0])
```
Comment (barbieri) [inline src/lib/common/sol-board-detect.c]: please filter to have only ".json", `strendswith(ent->d_name, ".json")`

File: tools/build/Makefile.vars [hunk 3]
```diff
@@ -335,8 +335,8 @@ $(call add-template,$(FLOW_METATYPE_BUILTINS_H_IN),$(FLOW_METATYPE_BUILTINS_H))
 NODE_TYPE_SCHEMA := $(top_srcdir)data/schemas/node-type-genspec.schema
 NODE_TYPE_SCHEMA_DEST := $(build_flowdatadir)schemas/node-type-genspec.schema
 
-BOARD_DETECT := $(top_srcdir)data/jsons/board_detect.json
-BOARD_DETECT_DEST := $(build_datadir)board_detect.json
+BOARD_DETECT := $(top_srcdir)data/jsons/99_board_detect.json
+BOARD_DETECT_DEST := $(build_datadir)/boards/99_board_detect.json
```
Comment (barbieri) [inline tools/build/Makefile.vars]: last one is always bad, usually people call this "50-default.json", so we know it is the default we provide and sits right in the "middle" of the range. People can add higher or lower priority versions as they see fit.

Comment (mbelluzzo) [inline tools/build/Makefile.vars]: Roger that

**Comment by mbelluzzo** [comment]
See V2

**Pull request closed without merging**